### PR TITLE
Add optional wait_for param in for entity creation

### PR DESCRIPTION
**Pull request by ereteog**
This PR enables to specify `wait_for` param when creating a resource in order to wait for availability of created resource (refresh index in case of ES store).

> / Close https://github.com/threatgrid/iroh/issues/2672

This PR adds an optional `wait_for` param in the creates routes that are generated in `src/ctia/http/routes/crud.clj`.
In crud tests, The creation is now performed with the `wait_for` param, and the the GET query is directly made after the POST query to ensure that the availability of the created resource is waited for.

<a name="qa">[§](#qa)</a> QA
============================

- Using a script, perform a  loop that repeats the following test 100 times: 
1. Create any entity with `wait_for=true` as query param
2. Retrieve the response ID and generate the short version of that ID (last part of the the url)
3. Retrieve that resource using the short ID
4. Check that the response status is 200

- Using a script, perform a loop that repeats the following test 100 times: 
1. Create any entity with`wait_for=false` as query param
2. Retrieve the response ID and generate the short version of that ID (last part of the the url)
3. Retrieve that resource using the short ID
4. Check the returned status: most of the status of the GET should be 404 (in some lucky cases some can be 200)

- Using a script, perform a loop that repeats the following test 100 times: 
1. Create any entity without the `wait_for` query param
2. Retrieve the response ID and generate the short version of that ID (last part of the the url)
3. Retrieve that resource using the short ID
4. Check the returned status: most of the status of the GET should be 404 (in some lucky cases some can be 200).

```
intern: When creating a resource, the query param `wait_for=true` could be used to deal with inconsistencies.
```

### CODE REVIEW

File: test/ctia/test_helpers/crud.clj [hunk 1]
```diff
@@ -54,15 +57,8 @@
...
-        (let [response (get (format "ctia/%s/%s" entity (:short-id record-id))
-                            :headers headers)
-              actor (:parsed-body response)]
-          (is (= 200 (:status response)))
-
-          (let [expected (assoc new-record :id (id/long-id record-id))]
-            (is (deep=
-                 expected
-                 record)))))
+        (is (= 200 get-status))
```
Comment (msprunck) [inline test/ctia/test_helpers/crud.clj]: Why did you move the request in the test of the POST operation?

File: test/ctia/test_helpers/crud.clj [hunk 2]
```diff
@@ -112,14 +106,41 @@
...
+          (is (= 200 (get-status true))
+              "Create queries should wait for index refresh when wait_for is true")
+          ;; we trigger next 404 tests twice because the refresh could occur between the first POST / GET sequence.
+          (is (or (= 404 (get-status false))
+                  (= 404 (get-status false)))
+             "Create queries should not wait for index refresh when wait_for is false")
+          (testing "Configured ctia.store.es.default.refresh value is applied when wait_for is not specified"
+            (if (= "false" (get-in properties [:ctia :store :es :default :refresh]))
+              (is (or (= 404 (get-status nil))
+                      (= 404 (get-status nil))))
```
Comment (yogsototh) [inline test/ctia/test_helpers/crud.clj]: same here.

**Comment by hshahman** [comment]
Verified in rel-v1.33
